enhancements/authentication/scc-pinning.md

```diff
@@ -0,0 +1,327 @@
+---
+title: scc-pinning
+authors:
+  - "@stlaz"
+reviewers:
+  - "@s-urbaniak"
+  - "@deads2k"
+  - "@sttts"
+approvers:
+  - "@deads2k"
+  - "@sttts"
+api-approvers:
+  - "@deads2k"
+  - "@sttts"
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+see-also:
+  - "/enhancements/this-other-neat-thing.md"
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# SCC pinning
+
+## Summary
+
+Many of the core workloads often rely on running with a specific Security Context Constraint (SCC).
+However, it ever so often happens that a customer brings an SCC of their own or a 3rd party product.
+Since SCCs are a complex construct, such a custom SCC often happens to be misshapen and certain fields
+may cause workloads to break if they depend on SCC defaulting their security contexts.
+
+## Motivation
+
+In order to prevent the core workloads from breaking by custom SCCs that might prevent them
+from running, introduce a system of pinning a specific SCC to a workload.
+
+### Goals
+
+1. Allow pinning one of the core SCCs provided with the OpenShift platform to be pinned
+   to a workload
+
+### Non-Goals
+
+1. Allow pinning a random SCC to a workload
+
+## Proposal
+
+### User Stories
+
+1. I am an owner of a core OpenShift workload and I want to be able to hint which
+   SCC should the SCC admission use in order to evaluate my workload's security context
+
+### API Extensions
+
+The SCC admission is currently using the "openshift.io/scc" annotation to denote which
```

Comment:
this could be potentially racy. shouldn't we have a separate dedicated annotation which expresses the intent to use an SCC?